addons/hr_expense/views/hr_expense_views.xml

```diff
@@ -53,27 +53,27 @@
                     <field name="product_has_cost" column_invisible="True"/>
                     <field name="date" optional="show" readonly="not is_editable"/>
                     <field name="product_id" optional="hide" readonly="not is_editable" required="1"/>
-                    <field name="name" readonly="not is_editable"/>
+                    <field name="name" readonly="not is_editable and state != 'approved'"/>
```

Comment:
After your change this field (and the others) are always editable (not readonly) when `state == 'approved'` (no matter the value of `is_editable`).
But `is_editable` also depends on user groups. Could this not cause an issue?